source/lib/vagrant-openstack-provider/client/neutron.rb

```diff
@@ -25,7 +25,11 @@ def get_all_networks(env)
       end
 
       def get_subnets(env)
-        subnets_json = get(env, "#{@session.endpoints[:network]}/subnets")
+        network_response = get(env, "#{@session.endpoints[:network]}")
+        network_resources = JSON.parse(network_response)['resources']
+        subnets_resource = network_resources.find { |x| x['name'] == 'subnet' }
+        subnets_url = subnets_resource['links'].find { |x| x['rel'] == 'self' }['href']
```

Comment:
Those 4 lines are repeated (only twice in your code, but if we do this change we have to do it for all the others OpenStack API calls (nova, cinder, glance and heat).

We should create a method for that. Probably in `http_utils.rb`. Something like

```ruby
def get_endpoint_url(service, object)
    endpoints = get(env, "#{@session.endpoints[service]}")
    resources = JSON.parse(endpoints)['resources']
    resource = resources.find { |x| x['name'] == object }
    subnets_resource['links'].find { |x| x['rel'] == 'self' }['href']
end
```

I don't ask you to change all the others API calls, we can do it as a second step. But if you are willing to do it, it would be awesome by the way.

Also, you can rebase your branch, the issue about `public_suffix` is fixed.